libsolidity/parsing/Parser.cpp

```diff
@@ -1313,11 +1313,32 @@ ASTPointer<InlineAssembly> Parser::parseInlineAssembly(ASTPointer<ASTString> con
 
 	expectToken(Token::Assembly);
 	yul::Dialect const& dialect = yul::EVMDialect::strictAssemblyForEVM(m_evmVersion);
+	ASTPointer<vector<ASTPointer<ASTString>>> dialectFlags;
 	if (m_scanner->currentToken() == Token::StringLiteral)
 	{
-		if (m_scanner->currentLiteral() != "evmasm")
-			fatalParserError(4531_error, "Only \"evmasm\" supported.");
+		std::string const& dialectString = m_scanner->currentLiteral();
+		auto it = dialectString.begin();
+		auto end = dialectString.end();
+		auto dialectStart = it;
+		while (it != end && !isspace(*it))
+			++it;
+
 		// This can be used in the future to set the dialect.
+		if (string(dialectStart, it) != "evmasm")
+			fatalParserError(4531_error, "Only \"evmasm\" supported.");
```

Comment:
I'd actually like to just parse the entire string as string and syntax-/type-check it later... but we can't really do that, since we need the dialect to confirm the dialect to actually parse the assembly below...

Conversely, I could move all the flags checks back here to the parser...